### ISSUE TITLE
Enable HDR with ARCore on Galaxy S8+ results in weird colors on camera background

### ISSUE THREAD

**Issue by NicolasWipon** [google-ar/arcore-unity-sdk#244]
Hi,

I have a weird bug when building an ARCore project on Samsung Galaxy S8+ or S8 with HDR enabled in Graphic Settings. The pixels colors from the camera are wrong, it looks like the green and blue values are shifted (green becomes blue and vice versa), as you can see on the 2 screenshots below.

Here is how to reproduce the issue : 
- Open an empty Unity Project and import ARCore package version 1.2.0
- Drag'n'drop "ARCore Device prefab" in the default scene (and delete the default "Main Camera").
- Go to Edit>Project Settings>Graphics
- Uncheck "Use Defaults" for Low, Medium and High sections, and check "Use HDR"
- Build and run on a Samsung Galaxy S8+ or S8 with Android v. 8.0.0 and you can see the problem with the colors of the camera feed.

The issue can be reproduced with Unity 2018.1 and 2017.3.
On Samsung Galaxy S7 and S7 Edge, it works well wih Android 7.0
The color problem only affects the camera feed, and not the 3d models we place on the scene.

![color_good](https://user-images.githubusercontent.com/39804538/40979842-441e3138-68d7-11e8-83da-07a89fd46286.jpg)
![color_altered](https://user-images.githubusercontent.com/39804538/40979846-475d575c-68d7-11e8-9eaa-fc25dad38526.jpg)

**Comment by NicolasWipon** [comment]
Hello,

Thanks for your answer.
Yes, it happens everytime. Have you enabled HDR in the Graphic Settings and on your Camera component ?
Maybe it is related to the smartphone model. The S8+ I tested on is SM-G955F. When I use an other way to get camera background (WebcamTextures or Vuforia), I don't have this issue, but when I use ARCore, I have it systematically.

**Comment by small-potatoes** [comment]
We are still aware of this issue and working with Unity to correct it.  Thank you for your patience.

**Comment by nedd-bastien** [comment]
Same issue here on a S8 with HDR, project is very close to deadline so I hope a correction is available soon.

**Comment by radishface** [comment]
Any updates regarding the issue?

**Comment by tdmowrer** [comment]
Unity engineer here. We are not able to reproduce this in 2017.4.8f1, 2018.1.1f1, or an unreleased 2019.1. We tested using [v1.5.0](https://github.com/google-ar/arcore-unity-sdk/releases) of the ARCore SDK for Unity.

@NicolasWipon does this still reproduce for you with ARCore 1.5? If so, could you [submit a bug report](https://unity3d.com/unity/qa/bug-reporting) and attach a project that reproduces the issue?

**Comment by NicolasWipon** [comment]
Hi tdmowrer,

Yes I still have this issue with ARCore v1.5.0 and Unity 2018.2.9f1 on Galaxy S8+ with Android 8.0.0 and "HDR" enabled in GraphicsSettings.

I am submitting a bug report with a project I just created this afternoon (there is just ARCore package and a SampleScene with "ARCore device" prefab).

Here are 2 screenshots taken today : the first one with the default Camera app of the device, showing the true colors, and the second one with the ARCore camera feed, showing the wrong colors.

I can't reproduce the issue when I use Vuforia instead of ARCore with the same configuration. 
![bug_arcore_true_colors](https://user-images.githubusercontent.com/39804538/49300402-6fb89280-f4c2-11e8-9c53-c4f5864af8d6.jpg)
![bug_arcore_wrongcolors](https://user-images.githubusercontent.com/39804538/49300418-78a96400-f4c2-11e8-8dbc-8e7c79f2f734.jpg)

**Comment by tdmowrer** [comment]
Thank you, @NicolasWipon . I was finally able to reproduce the issue on an S9+ (the GPU can vary slightly across devices, and since this seems to only affect certain models, that isn't too surprising).

It seems like the alpha channel is causing problems in certain cases, namely with this particular GPU variant and HDR enabled. Since we don't use the alpha channel for the background image (i.e., there is no blending), a simple fix would be to change the [`ARBackground.shader`](https://github.com/google-ar/arcore-unity-sdk/blob/master/Assets/GoogleARCore/SDK/Materials/ARBackground.shader#L48)'s fragment shader to look like this:
```glsl
void main()
{
    #ifdef SHADER_API_GLES3
    gl_FragColor = vec4(texture(_MainTex, textureCoord).xyz, 1);
    #else
    gl_FragColor = vec4(textureExternal(_MainTex, textureCoord).xyz, 1);
    #endif
}
```
That fixes the issue on the S9 I've been testing with. Could you let me know if that works for you?

Cheers,
Tim

**Comment by NicolasWipon** [comment]
Hi,

Yes, it works ! Thank you very much for the fix !

**Comment by small-potatoes** [comment]
This has been fixed in the [ARCore SDK for Unity version 1.7](https://github.com/google-ar/arcore-unity-sdk/releases) release.
